Add protobuf support

MordFustang21 opened the pull request:
Release Notes:

- Added protobuf syntax highlighting ([#5160](https://github.com/zed-industries/extensions/issues/146)).

MordFustang21 committed:
languages: add protobuf support

SomeoneToIgnore commented:
Looks much better now, thank you 🚀 
<img width="957" alt="image" src="https://github.com/zed-industries/zed/assets/2690773/7f1025e9-8c61-4bb6-8107-1f83188a0fa8">


I think we would need to add a bit more:

* `optional` keyword highlights 
* `oneof variant` highlighting also asks for more colors
* `outline.scm` so a person could navigate over the next with `cmd-shift-o` menu
* * ? maybe, add some slight color change to the field names, so they do not mix with the unknown identifiers
* and a small formatting nit the CI complained about needs to be fixed

maxdeviant commented on Cargo.toml:
```diff
@@ -162,6 +162,7 @@ tree-sitter-nu = { git = "https://github.com/nushell/tree-sitter-nu", rev = "26b
 tree-sitter-vue = {git = "https://github.com/zed-industries/tree-sitter-vue", rev = "6608d9d60c386f19d80af7d8132322fa11199c42"}
 tree-sitter-uiua = {git = "https://github.com/shnarazk/tree-sitter-uiua", rev = "9260f11be5900beda4ee6d1a24ab8ddfaf5a19b2"}
 tree-sitter-zig = { git = "https://github.com/maxxnino/tree-sitter-zig", rev = "0d08703e4c3f426ec61695d7617415fff97029bd" }
+tree-sitter-proto = {git = "https://github.com/rewinfrey/tree-sitter-proto", rev = "36d54f288aee112f13a67b550ad32634d0c2cb52"}
```
Can we keep these tree-sitter entries sorted alphabetically?

maxdeviant commented on crates/zed/Cargo.toml:
```diff
@@ -143,6 +143,7 @@ tree-sitter-nu.workspace = true
 tree-sitter-vue.workspace = true
 tree-sitter-uiua.workspace = true
 tree-sitter-zig.workspace = true
+tree-sitter-proto.workspace = true
```
Ditto here, let’s try to keep these in alphabetical order.

MordFustang21 replied:
You got it. I initially did that everywhere but I think I saw the entries here weren't sorted so I thought maybe you guys were just appending. I'll get that fixed.

maxdeviant replied:
I think they're _mostly_ sorted at the moment, but it's possible that some aren't entirely in order 😄

mikayla-maki commented on crates/zed/src/languages/proto/folds.scm:
```diff
@@ -0,0 +1,5 @@
+[
```
Zed does not support a `folds` query

mikayla-maki commented on crates/zed/src/languages/proto/highlights.scm:
```diff
@@ -0,0 +1,62 @@
+
+(int_lit) @number
+
+(float_lit) @float
```
We don't support `@float`

mikayla-maki reviewed (changes requested):
Thank you so much for adding this language! I've been meaning to do this for ages 😅. 

I added a few small things, and agree with my colleagues about the alphabetizing, highlighting, and outline file. If you'd like to drop by our channels or fireside hacks I'd be happy to help build the outline file. It would be great to get this one merged in :D

MordFustang21 commented:
I've addressed most of the above comments. There are still some outstanding issues though that I'm not quite sure how to tackle. 
1. `oneof variant highlighting also asks for more colors`
2. `? maybe, add some slight color change to the field names, so they do not mix with the unknown identifiers`

@mikayla-maki I added an outline file with some queries that felt most relevant. Happy to pair on it sometime if you feel there's more to add.

mikayla-maki commented:
Thank you for this PR! :D

SomeoneToIgnore commented:
Let's add a new doc entry into https://github.com/zed-industries/zed/tree/main/docs/src/languages list and merge this 🎉

MordFustang21 commented:
Added doc file

SomeoneToIgnore reviewed (approved):
I think it's still a bit odd to have the fields as defaults, but since I have no good proposal on how to capture it, let's move on.

Thank you for taking a stab at this, now Zed repo is better highlighted.
